src/search/landmarks/landmark_graph.h

```diff
@@ -55,104 +57,83 @@ class LandmarkNode {
-                if (state[fact.var].get_value() == fact.value) {
-                    return true;
-                }
-            }
-            return false;
-        } else { // conjunctive or simple
-            for (const FactPair &fact : facts) {
-                if (state[fact.var].get_value() != fact.value) {
-                    return false;
-                }
-            }
-            return true;
-        }
-    }
+    bool is_true_in_state(const State &state) const;
 };
 
 using LandmarkSet = std::unordered_set<const LandmarkNode *>;
 
 class LandmarkGraph {
 public:
+    /*
+      TODO: get rid of this by removing get_nodes() and instead offering
+      functions begin() and end() with an iterator class, so users of the
+      LandmarkGraph can do loops like this:
+        for (const LandmarkNode &n : graph) {...}
+     */
     using Nodes = std::vector<std::unique_ptr<LandmarkNode>>;
-    // ------------------------------------------------------------------------------
-    // methods needed only by non-landmarkgraph-factories
+private:
+    int num_conjunctive_landmarks;
+    int num_disjunctive_landmarks;
 
-    LandmarkNode *get_lm_for_index(int) const;
-    LandmarkNode *get_landmark(const FactPair &fact) const;
+    utils::HashMap<FactPair, LandmarkNode *> simple_landmarks_to_nodes;
+    utils::HashMap<FactPair, LandmarkNode *> disjunctive_landmarks_to_nodes;
+    Nodes nodes;
+    const TaskProxy &task_proxy;
 
-    // ------------------------------------------------------------------------------
-    // methods needed by both landmarkgraph-factories and non-landmarkgraph-factories
+    void remove_node_occurrences(LandmarkNode *node);
+public:
+    // needed only by landmarkgraph-factories.
+    explicit LandmarkGraph(const TaskProxy &task_proxy);
 
-    inline const Nodes &get_nodes() const {
+    // needed by both landmarkgraph-factories and non-landmarkgraph-factories
+    const Nodes &get_nodes() const {
         return nodes;
     }
-
-    inline int number_of_landmarks() const {
+    // needed by both landmarkgraph-factories and non-landmarkgraph-factories
+    int get_num_landmarks() const {
         return nodes.size();
     }
-
-    // ------------------------------------------------------------------------------
-    // methods needed only by landmarkgraph-factories
-    explicit LandmarkGraph(const TaskProxy &task_proxy);
-
-    inline LandmarkNode &get_simple_lm_node(const FactPair &a) const {
-        assert(simple_landmark_exists(a));
-        return *(simple_lms_to_nodes.find(a)->second);
-    }
-    inline LandmarkNode &get_disj_lm_node(const FactPair &a) const {
-        // Note: this only works because every proposition appears in only one disj. LM
-        assert(!simple_landmark_exists(a));
-        assert(disj_lms_to_nodes.find(a) != disj_lms_to_nodes.end());
-        return *(disj_lms_to_nodes.find(a)->second);
+    int get_num_disjunctive_landmarks() const {
```

Comment:
This method belongs to the group of "used by factories". It should be moved below even if it better "fits" here because of the other similar methods. When we further refactor the code, this method will be part of the factory classes rather than the landmark graph class.

Reply:
I think in the current code, it better fits here. If it will be moved out of the class in the future, it will disappear from here anyway, right?

Reply:
Sure, they will eventually disappear, but until then, I would like to keep the information they provide. As said above, if rearranging/regrouping, then let's at least make sure that it is clear which "group" each method belongs to by adding some comment to each method.